### PR TITLE
Everybody team, Franchise folder rights

### PR DESCRIPTION
**Pull request by danil-topc**
- Add newly creating users to 'Everybody' team
- Create Franchises folders with read access for 'Everybody', read-write for members of a group
- Create RocketChat profile for newly created users
- Refactor RocketChat object to global flask object

### CODE REVIEW

File: src/web/backend/nextcloud/utils.py [hunk 1]
```diff
@@ -50,9 +50,7 @@ def get_group_folder(mount_point):
     Returns:
     """
     nxc = get_nextcloud()
-    group_folders = nxc.get_group_folders().data
+    group_folders = nxc.get_group_folders(mount_point=mount_point).data
```
Comment (matejak) [inline src/web/backend/nextcloud/utils.py]: For starters, please keep the mount point filtering in TEAP a separate function in - you can port it to Nextcloud API later.

**Comment by matejak** [comment]
Hello, it looks good - I will merge it as soon as the minor nextcloud API issue is addressed.

Concerning dependencies on "family" projects, I suggest going for the approach of implementing the functionality in TEAP if possible in a form that allows transferring it to any other project fast, and "cleaning" TEAP afterwards. Depending on other project PRs makes the development somehow less flexible.

File: src/web/backend/nextcloud/utils.py (hunk 1 above)
Comment (danil-topc) [reply src/web/backend/nextcloud/utils.py]: reverted it to filter in iteration as it was before

**Comment by danil-topc** [comment]
> Concerning dependencies on "family" projects, I suggest going for the approach of implementing the functionality in TEAP if possible in a form that allows transferring it to any other project fast, and "cleaning" TEAP afterwards. Depending on other project PRs makes the development somehow less flexible.
Good point, will mark such changes as `TODO` in TEAP and will fix them after.

**Comment by matejak** [comment]
Great, thanks!

**Pull request merged**